webapp/backend/src/handlers/api/videos.ts

```diff
@@ -0,0 +1,63 @@
+import { Request, Response, Router } from "express";
+import fs from "fs";
+import allVideos from "../../videoData";
+import path from "path";
+
+// Initialise router object
+const demoVideos = Router();
+
+// Metadata for all videos
+demoVideos.get("/", (req: Request, res: Response) => {
+  res.json(allVideos);
+});
+
+// Metadata for a single video
+demoVideos.get("/:id/metadata", (req: Request, res: Response) => {
+  const id = parseInt(req.params.id, 10);
+  res.json(allVideos[id]);
+});
+
+// Endpoint for a single video to stream
+demoVideos.get("/video/:id", (req: Request, res: Response) => {
```

Comment:
We already have the parent `/videos` level which is the collection URI. We don't want to unnecessarily make the api endpoint too long. Let's follow the REST api naming convention below and go with `/videos/{id}` for a particular video instead of `/videos/video/{id}`

https://restfulapi.net/resource-naming/

Reply:
please go over the api conventions as well for future use cases also.

Reply:
> please go over the api conventions as well for future use cases also.

Sure. Thanks for the additional resource.